File: Skeleton code for AI/Python/send_DTO.py
import json
#### DTO Classes
class InputAction:
    def __init__(self, actionType, properties):
        self.ActionType = actionType
        self.Properties = properties
        
        
    def toJSON(self):
        return json.dumps(self, default=lambda obj: obj.__dict__)
class Action:
    def __init__(self, x = 0, y = 0, cardid = 0, amount = 0):
        self.X = x
        self.Y = y
        self.Cardid = cardid
        self.Amount = amount

File: Skeleton code for AI/Cpp/SendDTO.cpp
#include <string>
#include <list>
#include "ReceiveDTO.cpp"
using namespace std;

class Action {
public:
	int x;
	int y;
	int cardId;
	int amount;
	Action(int x = 0, int y = 0, int cardId = 0, int amount = 0) {
		this->x = x;
		this->y = y;
		this->cardId = cardId;
		this->amount = amount;
	}
};

class InputAction {
public:
	string actionType;
	list<Action> properties;
	InputAction(string actionType, list<Action> properties) {
		this->actionType = actionType;
		this->properties = properties;
	}

	/*string toString() {}*/
};


